src/transformers/generation/utils.py

```diff
@@ -698,9 +698,12 @@ def _get_candidate_generator(
         Returns the candidate generator to be used in `assisted_generation`
         """
         if generation_config.prompt_lookup_num_tokens is not None:
-            candidate_generator = PromptLookupCandidateGenerator(
+            candidate_generator_params = dict(
```

Comment:
Instead of creating a dict here, let's:
a) pass keyword arguments to `PromptLookupCandidateGenerator` (as before the PR)
b) default `max_matching_ngram_size` in `PromptLookupCandidateGenerator` to `None`, and set it to the original default value in `__init__` if it is `None`.

This pushes complexity away from `generate` and into `PromptLookupCandidateGenerator` :)

Reply:
Thanks for the comment! I switched back to keyword arguments.